structopt-derive/src/lib.rs

```diff
@@ -227,6 +228,52 @@
 //!     Quux
 //! }
 //! ```
+//!
+//! ## Custom string parsers
+//!
+//! If the field type does not have a `FromStr` implementation, or you would
+//! like to provide a custom parsing scheme other than `FromStr`, you may
+//! provide a custom string parser using `parse(...)` like this:
+//!
+//! ```ignore
+//! use std::num::ParseIntError;
+//! use std::path::PathBuf;
+//!
+//! fn parse_hex(src: &str) -> Result<u32, ParseIntError> {
+//!     u32::from_str_radix(src, 16)
+//! }
+//!
+//! #[derive(StructOpt)]
+//! struct HexReader {
+//!     #[structopt(short = "n", parse(try_from_str = "parse_hex"))]
+//!     number: u32,
+//!     #[structopt(short = "o", parse(from_os_str))]
+//!     output: PathBuf,
+//! }
+//! ```
+//!
+//! There are four kinds custom string parsers:
+//!
+//! | Kind              | Signature                             | Default                         |
+//! |-------------------|---------------------------------------|---------------------------------|
+//! | `from_str`        | `fn(&str) -> T`                       | `::std::convert::From::from`    |
+//! | `try_from_str`    | `fn(&str) -> Result<T, E>`            | `::std::str::FromStr::from_str` |
```

Comment:
That's a bit confusing that `::std::str::FromStr::from_str` is the default for `try_from_str` and not `from_str` but I can't find a better idea.